src/runtime_src/core/pcie/windows/device_windows.cpp

```diff
@@ -697,6 +708,16 @@ struct devinfo
         //sample strings: xilinx_u250_GOLDEN, xilinx_u250_gen3x16_base, xilinx_u250_xdma_201830_3
         return (shell.find("GOLDEN") != std::string::npos) ? false : true;
       }
+    case key_type::mac_addr_first:
+      return std::string(info.MacAddrFirst);
+    case key_type::mac_contiguous_num:
+      return info.MacContiguousNum;
+    case key_type::mac_addr_list:
+      vec.push_back(std::string(info.MacAddr0));
+      vec.push_back(std::string(info.MacAddr1));
+      vec.push_back(std::string(info.MacAddr2));
+      vec.push_back(std::string(info.MacAddr3));
+      return std::vector<std::string>(vec);
```

Comment:
At a minimum scope the vector under the case or factor out into separate function.
Also why return a copy of the vector as opposed to just the vector itself?